test/gtest/ucp/test_ucp_request.cc

```diff
@@ -644,87 +638,101 @@ class test_proto_reset_atp : public test_proto_reset {
         ucp_lane_index_t num_lanes = ucp_ep_config(ep)->key.num_lanes;
 
         for (ucp_lane_index_t lane = 0; lane < num_lanes; ++lane) {
-            ucp_ep_get_lane(ep, lane)->iface->ops = m_ops[lane];
+            uct_ep_h uct_ep    = ucp_ep_get_lane(ep, lane);
+            uct_ep->iface->ops = m_ops[uct_ep];
         }
     }
 
+    static bool is_second_atp(uint8_t id)
+    {
+        static bool second = false;
+        bool res           = second && (id == UCP_AM_ID_RNDV_ATP);
+        second             = true;
+        return res;
+    }
+
+    static ucs_status_t uct_am_short(uct_ep_h ep, uint8_t id, uint64_t header,
+                                     const void *payload, unsigned length)
+    {
+        if (is_second_atp(id)) {
+            return UCS_ERR_NO_RESOURCE;
+        }
+
+        return m_ops[ep].ep_am_short(ep, id, header, payload, length);
```

Comment:
- Add a member var to count ATPs (so test works when running multiple variants/iterations)
- Remove is_second_atp to simplify:
```c
/* Pause on 2nd ATP message */
if ((id == UCP_AM_ID_RNDV_ATP) && (++m_atp_count == 2)) {
     return UCS_ERR_NO_RESOURCE;
}

return m_ops[ep].ep_am_short(ep, id, header, payload, length);
```